File: lcm.ts
/**
 * LCM Luxe TypeScript Runtime
 * Strongly-typed, immutable-input LCM computation suitable for Node.js
 * and browser bundles compiled with tsc, esbuild, or Vite.
 */

export function LCM(A: ReadonlyArray<number>): number {
  if (A.some((x: number) => x === 0)) {
    return 0;
  }

  const gcd = (a: number, b: number): number =>
    b === 0 ? a : gcd(b, a % b);

  const lcm = (a: number, b: number): number =>
    Math.abs(a * b) / gcd(a, b);

  return A.reduce((acc, val) => lcm(acc, val), 1);
}

// Sanity check when executed directly with ts-node or Deno.
if (import.meta.main) {
  const sample = [-50, 25, -45, -18, 90, 447];
  console.log(`LCM(${sample.join(", ")}) = ${LCM(sample)}`);
}

File: lcm.py
"""
LCM Luxe Python Runtime
Idiomatic, type-hinted implementation using math.gcd and functools.reduce.
"""

from math import gcd
from functools import reduce


def _pair_lcm(a: int, b: int) -> int:
    """Compute the least common multiple of two integers."""
    return abs(a * b) // gcd(a, b)


def LCM(values: list[int]) -> int:
    """
    Compute the least common multiple of an integer array.

    Parameters
    ----------
    values: list[int]
        A non-empty list of integers.

    Returns
    -------
    int
        The least common multiple. Returns 0 if any input is 0.
    """
    if any(x == 0 for x in values):
        return 0
    return reduce(_pair_lcm, values)


if __name__ == "__main__":
    sample = [-50, 25, -45, -18, 90, 447]
    print(f"LCM({sample}) = {LCM(sample)}")
